compiler/noirc_evaluator/src/ssa/opt/constant_folding.rs

```diff
@@ -242,10 +242,8 @@ impl<'brillig> Context<'brillig> {
             function.dfg.make_constant(FieldElement::one(), NumericType::bool());
 
         for instruction_id in instructions {
-            if !self.values_to_replace.is_empty() {
```

Comment:
We check `!mapping.is_empty()` internally in `Instruction::replace_values`

Reply:
Good point! I think I did it to avoid hitting `function.dfg[instruction_id]` but it seems that's just an array index, not a map lookup, so it shouldn't impact performance.